### PR TITLE
Update MFO API to use yaml instead of map

### PR DESCRIPTION
**Pull request by rewantsoni**
Creating this PR as a replacement for https://github.com/red-hat-storage/managed-fusion-agent/pull/58

### CODE REVIEW

File: controllers/managedfusionoffering_controller.go [hunk 1]
```diff
@@ -386,3 +392,16 @@ func pluginGetDesiredSubscriptionSpec(r *ManagedFusionOfferingReconciler) *opv1a
...
+
+// This function is a placeholder for offering plugin integration
+func pluginGetDesiredOfferingSpec(r *ManagedFusionOfferingReconciler) interface{} {
+	var offeringSpec interface{}
+
+	switch r.managedFusionOffering.Spec.Kind {
+	case v1alpha1.KindDataFoundation:
+		offeringSpec = dataFoundationSpec{}
+	case v1alpha1.KindDataFoundationClient:
+		offeringSpec = dataFoundationClientSpec{}
```
Comment (nb-ohad) [inline controllers/managedfusionoffering_controller.go:404]: Let's not have offering logic inside the controller package. You should assume that both the offering codes is going to move to the offering packages
```suggestion
		offeringSpec = dfGetOfferingSpecInstance() 
	case v1alpha1.KindDataFoundationClient:
		offeringSpec = dfcGetIfferingSpecInstance()
```

**Comment by nb-ohad** [comment]
@bindrad I believe this PR is ready to be merged

**Comment by bindrad** [comment]
@rewantsoni Can you rebase this PR?

**Comment by rewantsoni** [comment]
> @rewantsoni Can you rebase this PR?

Rebased

**Pull request merged**